src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -232,5 +269,21 @@ private IEnumerable<WorkloadId> GetUpdatableWorkloads()
 
             return workloads;
         }
+
+        private void RunInNewTransaction(Action<ITransactionContext> a)
```

Comment:
Is this the exact same code as in `WorkloadInstallCommand`?  If so, should this be moved up to the base class?

Reply:
I actually had the exact same thought. The only issue is that the strings are separate, so this is an annoyingly big change. I do think it would be cleaner to do it that way, though...